src/Neo/SmartContract/ContractBasicMethod.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ContractBasicMethod.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+namespace Neo.SmartContract
+{
+    /// <summary>
+    /// This class provides a guideline for basic methods used in the Neo blockchain, offering
+    /// a generalized interaction mechanism for smart contract deployment, verification, updates, and destruction.
+    /// </summary>
+    public record ContractBasicMethod
```

Comment:
```suggestion
    internal record ContractBasicMethod
```

Reply:
This will be used in plugins, internal will require to add those plugins as internally accessible.